numpy/lib/tests/test_nanfunctions.py

```diff
@@ -231,79 +231,77 @@ class MyNDArray(np.ndarray):
             assert_(res.shape == ())
 
 
-class TestNanFunctions_IntTypes:
-
-    int_types = (np.int8, np.int16, np.int32, np.int64, np.uint8,
-                 np.uint16, np.uint32, np.uint64)
+@pytest.mark.parametrize(
+    "dtype",
+    np.typecodes["AllInteger"] + np.typecodes["AllFloat"] + "O",
```

Comment:
`np.typecodes['AllFloat'][1:]` would remove `'e'`, but is not quite as robust.

Reply:
No worries, as of https://github.com/numpy/numpy/pull/19821/commits/fffcb6e2bd45bf423f626ebf77a2e3142de636c4 the tests seem to work fine with `np.float16`.